pkg/chunkenc/memchunk.go

```diff
@@ -1761,6 +1765,19 @@ func (e *entryBufferedIterator) StreamHash() uint64 { return e.pipeline.BaseLabe
 
 func (e *entryBufferedIterator) Next() bool {
 	for e.bufferedIterator.Next() {
+		// If this is a NoProcessingStreamPipeline, bypass processing
+		if e.isNoProcessingPipeline {
+			e.currLabels = e.pipeline.BaseLabels()
+			e.cur.Timestamp = time.Unix(0, e.currTs)
+			// E.Welch it's likely possible to avoid the copy here however
+			// there is always risk with unsafe copies and this bypass already goes
+			// a long way to reduce the work we used to do when processing
+			// structured meatdata in the previous NoOpPipeline
+			e.cur.Line = string(e.currLine)
+			e.cur.StructuredMetadata = logproto.FromLabelsToLabelAdapters(e.currStructuredMetadata)
```

Comment:
ISTM this is the only line which needs to do something different between this `if` and the normal case.  Is it worth having such a large diff?

Reply:
Thanks, I see now that the `LabelsResult` type is more complicated.